### ISSUE TITLE
Uncertain instances in MNIST

### ISSUE THREAD

**Issue by zhangabner** [zhangabner/ML#3]
The MNIST database of handwritten digits, has a training set of 60,000 examples, and a test set of 10,000 examples. It is a subset of a larger set available from NIST Special Database including handwritten of digits and upper and lower case, which are hand checked. The handwritten digit is a symbol which is difficult to make a mutually exclusive definition, and even difficult to decide what is the correct label. Since there is no definition, I propose a possible def-inition used in NIST: in the context of handwritten digits recognition, based on the shape of handwritten, choose the closest to the print version between 0-9.
There are 3 parties involved the decision of the label in NIST Special Database, the writers, the image processors, and the label checkers.
The writers are required to write the handwritten below the print symbols as Figure 1

 
Figure1 the writer d3462’s original form
The image processors do the image segmentation and as-sign a label based on the print symbols. They are involved, because some writers handwriting was not segmented cor-rectly, such as writer d3462, many handwritings’ the below part are missing. 

 
Figure2 the writer d3462’s handwritten digits’ segmentation is not correct. 
Figure1 the part of original form of writer d3462 shows the handwritings are close to the button line, which causes the wrong segmentation, and it also shows the handwritten of 2, which is different with d3462_15_00012.
The part of a symbol could be another symbol, as below shows part of 3 becomes to 2, part of 2 becomes 7.

 
Figure3 shows that different part of a handwritten could be another sym-bol.
The label checker notice that the image d3462_15_00012 is closer to 2, and change the label from 3 to 2. Because in the context of the original form, it is 3, and in the context of digital recognition in 0-9, it is closer to 2.
In MNIST, 3443 and 17229 are two examples, which are not segmented completely, it’s label is 2, but it is part of 3.

  3443 

17229
It is also true when the image is changed in other ways. For instance, when adversarial noise is added into the image, the label should not be the same.
At same time, label checkers could make mistake. For ex-ample, below shows 4 wrong labels:

2080 labeled as 3

54915 labeled as 4

30310 labeled as 5

 44960 labeled as 6
And sometimes it is impossible to give a correct label. For example, the given labels in below figure 4 are 1,4,4,7, but the labels could change based on different bias of different readers, and it could be difficult to reach an agreement on what’s the correct label. 

 
64201     25792       32450    4433
Figure 4 confusion labels
Another example, the given labels in below figure 5 are 4,4,4,4, but it is difficult to say they are handwritten digits.

 
60033     9715       27776     24922
Figure 5 confusion labels
The uncertain might be the best label for these images.
The uncertain labels in MNIST are Mid-objects label, un-known label and unclear label.
Unknown labels:

 12958 in the context of form is 9, but in a large con-text of 0-9 and a-z, it should be ‘g’, which is unknown in 0-9.

 23652 is labeled as 4 and some readers may think it is closer to 9, but in a large context of 0-9 and a-z, it should be ‘q’, which is unknown in 0-9.
Mid-label:

number 50598 labeled as 4, based on different reader’s bias, it could be 7 as figure4 shows.
Unclear label:

66571 labeled as 9
